classes/components/forms/context/PKPDateTimeForm.inc.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * @file classes/components/form/context/PKPDateTimeForm.inc.php
+ *
+ * Copyright (c) 2014-2020 Simon Fraser University
+ * Copyright (c) 2000-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPDateTimeForm
+ * @ingroup classes_controllers_form
+ *
+ * @brief A preset form for presenting date and time on the frontend
+ */
+namespace PKP\components\forms\context;
+use PKP\components\forms\FieldRadioInput;
+use \PKP\components\forms\FormComponent;
+
+define('FORM_DATE_TIME', 'dateTime');
+
+class PKPDateTimeForm extends FormComponent {
+	/** @copydoc FormComponent::$id */
+	public $id = FORM_DATE_TIME;
+
+	/** @copydoc FormComponent::$method */
+	public $method = 'PUT';
+
+	/**
+	 * Constructor
+	 *
+	 * @param $action string URL to submit the form to
+	 * @param $locales array Supported locales
+	 * @param $context \Context Journal or Press to change settings for
+	 */
+	public function __construct($action, $locales, $context) {
+		$this->action = $action;
+		$this->locales = $locales;
+		$currentDateTime = time();
+
+		$localizedOptions = []; // template for localized options to be used for date and time format
+		foreach ($this->locales as $key => $localeValue) {
+			$localizedOptions[$localeValue['key']] = $key;
+		}
+
+		$this->addGroup([
+				'id' => 'descriptions',
+				'label' => __('manager.setup.dateTime.descriptionTitle'),
+				'description' => __('manager.setup.dateTime.description'),
+			])
+			->addField(new FieldRadioInput('dateFormatShort', [
```

Comment:
I think that none of these fields are actually required, right? They all have fallbacks, so I think you can remove `isRequired` from each one.